hera_cal/smooth_cal.py

```diff
@@ -325,17 +383,20 @@ def __init__(self, calfits_list, flag_file_list=[], flag_filetype='h5', pick_ref
                     if ant in self.ext_flags[ff]:
                         self.flag_grids[ant][self.flag_time_indices[ff], :] += self.ext_flags[ff][ant]
 
-        # perform data quality checks
+        # perform data quality checks and flag thresholding
         self.check_consistency()
-        self.reset_filtering()
+        flag_threshold_and_broadcast(self.flag_grids, freq_threshold=freq_threshold,
+                                     time_threshold=time_threshold, ant_threshold=time_threshold)
 
         # pick a reference antenna that has the minimum number of flags (tie goes to lower antenna number) and then rephase
         if pick_refant:
             utils.echo('Now picking reference antenna(s)...', verbose=self.verbose)
             self.refant = pick_reference_antenna(self.gain_grids, self.flag_grids, self.freqs, per_pol=True)
             utils.echo('\n'.join(['Reference Antenna ' + str(self.refant[pol][0]) + ' selected for ' + pol + '.' 
                                   for pol in sorted(list(self.refant.keys()))]), verbose=self.verbose)
-            self.rephase_to_refant()
+
+        self.reset_filtering()  # initializes self.filtered_gain_grids and self.filtered_flag_grids
```

Comment:
Is there a use case for resetting the filtering, but then doing nothing if refant is not set? Put another way, can the work done inside of `reset_filtering` actually be done in `rephase_to_refant`? It would make it easier for the user to track what's going on, and potentially avoid duplicating the gains if rephasing isn't actually done.

Reply:
The rephasing happens in place (and we want it to keep happening after the filtering so that the reference antennas stays the reference antenna). So it's useful that the rephasing function applies to self.filtered_gain_grids instead of self.gain_grids. I could explicitly initialize the copies in the __init__, but 

The doubling of gains is unavoidable if we want to maintain the resetting functionality. I could get rid of that functionality to save memory if we wanted...

Reply:
yeah, let's kill the second copy... If someone is doing interactive work and wants to keep it around, they can explicitly deepcopy it.

Reply:
Ok, done.

Reply:
Great, thanks!